### PR TITLE
Migrate from UnsafePointer<Void> to UnsafeRawPointer.

### PR DESCRIPTION
**Pull request by atrick**

### CODE REVIEW

**Comment by gribozavr** [comment]
LGTM.

**Comment by atrick** [comment]
There are a few test case failures that I still need to investigate. Such as:
func takeDSOHandle(_ handle: UnsafeMutableRawPointer = #dsohandle) { }
                                                       ^~~~~~~~~~
These contextual type purposes cannot fail with a conversion type specified!
UNREACHABLE executed at /s/sptr/swift/lib/Sema/CSDiag.cpp:3525!

File: stdlib/public/core/UnsafeRawPointer.swift.gyb [hunk 1]
```diff
@@ -27,6 +27,13 @@ public struct Unsafe${Mutable}RawPointer : Strideable, Hashable, _Pointer {
   /// Implements conformance to the public protocol `_Pointer`.
   public let _rawValue: Builtin.RawPointer
 
+  // Construct ${a_Self} from another ${a_Self}.
+  // FIXME: Why is this necessary?
+  @_transparent
+  public init(_ other: Unsafe${Mutable}RawPointer) {
+    self._rawValue = other._rawValue
```
Comment (gribozavr) [inline stdlib/public/core/UnsafeRawPointer.swift.gyb]:
```swift
self = other
```
?

**Comment by atrick** [comment]
@swift-ci Please smoke test OS X.

**Comment by atrick** [comment]
Depends on corelibs-libdispatch PR:
https://github.com/apple/swift-corelibs-libdispatch/pull/120

Depends on corelibs-foundation PR:
https://github.com/apple/swift-corelibs-foundation/pull/472

**Comment by atrick** [comment]
build-toolchain passes on both darwin and linux.

**Pull request merged**